agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -126,8 +173,35 @@ CompletableResultCode internalSendByConnectionString(
     return telemetryPipeline.send(byteBuffers, connectionString, listener);
   }
 
-  List<ByteBuffer> encode(List<TelemetryItem> telemetryItems) throws IOException {
+  private static TelemetryItem createOtelResourceMetric(
+      Map<String, String> existingTags, String connectionString)
+      throws UnsupportedEncodingException {
+    MetricTelemetryBuilder builder = MetricTelemetryBuilder.create(_OTELRESOURCE_, 0);
+    // this is needed in order to stamp iKey onto the telemetry item during serialization
+    builder.setConnectionString(connectionString);
+    builder.addTag(
+        ContextTagKeys.AI_CLOUD_ROLE.toString(),
+        existingTags.get(ContextTagKeys.AI_CLOUD_ROLE.toString()));
+    builder.addTag(
+        ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString(),
+        existingTags.get(ContextTagKeys.AI_CLOUD_ROLE_INSTANCE.toString()));
+    builder.addTag(
+        ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString(),
+        existingTags.get(ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString()));
+
+    // add attributes from OTEL_RESOURCE_ATTRIBUTES
+    for (Map.Entry<String, String> entry : OTEL_RESOURCE_ATTRIBUTES.entrySet()) {
+      // Attributes specified via otel.resource.attributes follow the W3C Baggage spec and
+      // characters outside the baggage-octet range are percent encoded
+      // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#specifying-resource-information-via-an-environment-variable
+      builder.addProperty(
+          entry.getKey(),
+          URLDecoder.decode(entry.getValue(), StandardCharsets.UTF_8.displayName()));
+    }
```

Comment:
I realized, better to do the URLDecoder once during the initialization of `OTEL_RESOURCE_ATTRIBUTES`